### PR TITLE
feat(api): Included support for API query for additional parameters

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves - Jira - 318

### 

# Explain the feature/fix
This PR provides the API support for fetch users by rhatUUID, name, uid and all

## Does this PR introduce a breaking change
No

## Screenshots
N/A

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/user-service/src/resolver.ts [hunk 1]
```diff
@@ -5,94 +5,98 @@ import { UserGroupAPIHelper } from './helpers';
 
 export const UserGroupResolver = {
   Query: {
-    getUser(root: any, args: any, ctx: any) {
-      return User.findOne({ uid: args.uid })
-        .then(data => data)
-        .catch(err => err);
+    getUser ( root: any, args: any, ctx: any ) {
```
Comment (deshmukhmayur) [inline packages/user-service/src/resolver.ts:8]: Also, once we have the new query for `getUsersBy`, we can deprecate this as the new query can do the same job.

Comment (riginoommen) [reply packages/user-service/src/resolver.ts]: Updated

File: packages/user-service/src/e2e/user.spec.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ const query = `
         ...userType
     }
   }
-  query GetUser($uid: String!) {
-    getUser(uid: $uid) {
+  query GetUserBy($uid: String!) {
+    getUserBy(uid: $uid) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/user-service/src/e2e/user.spec.ts]: Just updated the missing code

File: packages/user-service/src/e2e/user.spec.ts [hunk 3]
```diff
@@ -44,112 +46,135 @@ const query = `
...
+        expect( res.body.data.getUsersBy[ 0 ] ).toHaveProperty( 'uid' );
+        expect( res.body.data.getUsersBy[ 0 ] ).toHaveProperty( 'name' );
+        expect( res.body.data.getUsersBy[ 0 ] ).toHaveProperty( 'title' );
+      } )
+      .end( ( err, res ) => {
+        done( err );
+      } );
+  } );
+
+  it( 'should get Users by uid', ( done ) => {
```
Comment (deshmukhmayur) [inline packages/user-service/src/e2e/user.spec.ts:106]:
```suggestion
  it( 'should list all Users', ( done ) => {
```

File: packages/user-service/src/typedef.graphql [hunk 4]
```diff
@@ -127,8 +127,8 @@ type Mutation  {
 }
 
 type Query  {
-  # Fetches specific User by id
-  getUser(uid: String!): UserType
+  # Fetches specific User by uid, rhatUUID, apiRole, name
+  getUsersBy( _id:String, uid: String, rhatUUID: String, apiRole: String, name: String): [UserType]
```
Comment (deshmukhmayur) [inline packages/user-service/src/typedef.graphql:131]: Can you change the _id to id? We shouldn't expose the _id as an external parameter/input.

```suggestion
  getUsersBy( id: String, uid: String, rhatUUID: String, apiRole: String, name: String): [UserType]
```

**Pull request merged**